ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -206,9 +210,9 @@ constructor(
   private suspend fun updateCameraPositionOnLocationChange() {
     getLocationUpdates()
       .map { it.toCoordinates() }
-      .withIndex()
-      .collect { (index, coordinates) ->
-        if (index == 0) {
+      .collect { coordinates ->
+        if (isLocationLockReset) {
+          isLocationLockReset = false
```

Comment:
The way it's written now kit's not immediately clear what we're trying to do something different on the first emission. Would it be more idiomatic to still use `withIndex()` and to reset the stream each time to location updates are enabled (maybe via flatMapLatest?)

Reply:
I wasn't sure how to reset the stream. Will check once again.

Reply:
```kt
  locationLockEnabled
    .flatMapLatest { enabled -> getLocationUpdates().filter { enabled } }
    .withIndex()
    .collect { ...
    }
```

Reply:
Thanks for the suggestion! This also gets rid of the boolean bit